src/Providers/ProvidersPage.tsx

```diff
@@ -200,28 +215,87 @@ export const ProvidersPage = ({ namespace, kind }: ProvidersPageProps) => {
             </ToolbarToggleGroup>
           </ToolbarContent>
         </Toolbar>
-
-        {loaded && error && <Errors />}
-        {!loaded && <Loading />}
-        {loaded && !error && (
-          <TableView<MergedProvider>
-            entities={providers.filter(
-              createMetaMatcher(selectedFilters, fields),
-            )}
-            allColumns={fields}
-            visibleColumns={fields.filter(({ isVisible }) => isVisible)}
-            aria-label={t('Providers')}
-            Row={ProviderRow(kind)}
-          />
-        )}
+        <TableView<MergedProvider>
+          entities={filteredProviders}
+          allColumns={fields}
+          visibleColumns={fields.filter(({ isVisible }) => isVisible)}
+          aria-label={t('Providers')}
+          Row={ProviderRow}
+        >
+          {[
+            !loaded && <Loading />,
+            loaded && error && <ErrorState />,
+            loaded && !error && providers.length == 0 && <NoResultsFound />,
+            loaded &&
+              !error &&
+              filteredProviders.length === 0 &&
+              providers.length > 0 && (
+                <NoResultsMatchFilter clearAllFilters={clearAllFilters} />
+              ),
```

Comment:
Can we move the checks to some consts ?
for example:
``` ts
const errorFetchingData =  loaded && error;
const noResoults =  loaded && !error && providers.length == 0;
const noMatchingResoults = loaded && !error &&
    filteredProviders.length === 0 && providers.length > 0;

```
```suggestion
            !loaded && <Loading />,
            errorFetchingData && <ErrorState />,
            noResoults && <NoResultsFound />,
            noMatchingResoults && <NoResultsMatchFilter clearAllFilters={clearAllFilters} />
              ),
```